src/value-utils.ts

```diff
@@ -45,4 +45,60 @@ export const durationFromTimestamp = (timestamp: number): string => {
   return str.trim();
 };
 
+export const millisecondsFromDuration = (duration: string): number => {
+  try {
+    const parts = duration
+      .trim()
+      .split(/\s+/)
+      .map((p) => p.match(/^(\d+)([wdhms])$/));
+    return parts.reduce((sum, p) => {
+      if (p && p.length >= 3) {
+        const digit = p[1];
+        const unit = p[2];
+
+        return sum + parseInt(digit, 10) * units[unit];
+      }
+      return sum;
+    }, 0);
+  } catch (ignored) {
+    // Invalid duration format
+    return 0;
+  }
+};
+
+export const formatDate = (timestamp: number): string => {
+  const formatter = new Intl.DateTimeFormat('en', {
+    year: 'numeric',
+    month: '2-digit',
+    day: '2-digit',
+  });
+  const parts: Record<string, string> = {};
+
+  formatter
+    .formatToParts(timestamp)
+    .filter((item) => ['year', 'month', 'day'].includes(item.type))
+    .forEach(({ type, value }) => (parts[type] = value));
+
+  return `${parts.year}-${parts.month}-${parts.day}`;
+};
+
+export const formatTime = (timestamp: number, includeSeconds = false): string => {
+  const formatter = new Intl.DateTimeFormat('en', {
+    hour: '2-digit',
+    hourCycle: 'h24',
+    minute: '2-digit',
+    second: includeSeconds ? 'numeric' : undefined,
```

Comment:
```suggestion
    second: 'numeric',
```

Because we test `includeSeconds` later anyway